code/script.js

```diff
@@ -225,101 +232,164 @@ const generateBoard = () => {
 // Randomly select a person from the characters array and set as the value of the variable called secret
 const setSecret = () => {
   secret = charactersInPlay[Math.floor(Math.random() * charactersInPlay.length)]
+  console.log('secret person selected')
+  console.log(secret)
 }
 
 // This function to start (and restart) the game
 const start = () => {
+  console.log('start funktion is called!')
   // Here we're setting charactersInPlay array to be all the characters to start with
   charactersInPlay = CHARACTERS
   // What else should happen when we start the game?
+  // set the secret person -> invoke set Secret funktion
+  setSecret()
+  //show the board on the screen -> invoking generateboard funktion. 
+  generateBoard()
 }
 
 // setting the currentQuestion object when you select something in the dropdown
 const selectQuestion = () => {
-  const category = questions.options[questions.selectedIndex].parentNode.label
-
+  const category = questions.options[questions.selectedIndex].parentNode.label  
+  const value = questions.options[questions.selectedIndex].value
   // This variable stores what option group (category) the question belongs to.
   // We also need a variable that stores the actual value of the question we've selected.
-  // const value =
-
+  
   currentQuestion = {
     category: category,
-    // value: value
+    value: value,
   }
+  console.log('question asked', currentQuestion)
 }
 
 // This function should be invoked when you click on 'Find Out' button.
+let count = 0
 const checkQuestion = () => {
-  const { category, value } = currentQuestion
+  count+= 1
+  guessCounter.innerHTML = " "
+  guessCounter.innerHTML += `<p> Guesses made: ${count}</p>`
 
+  const { category, value } = currentQuestion
+  console.log('Let´s findout!')
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
 
+  if (category === 'hair' || category === 'eyes') {
+    if(value === secret.hair || value === secret.eyes){
+      keep = true
+      filterCharacters(keep)
+    }
+    else{
+      keep = false
```

Comment:
I don't think `keep = false` is being used anywhere because when you call `filterCharacters` you don't take keep as an argument, so it could be deleted here and on line 293